optimism/Mechanics.py

```diff
@@ -246,7 +246,9 @@ def compute_initial_state():
 ######
     
     
-def create_mechanics_functions(functionSpace, mode2D, materialModel, pressureProjectionDegree=None):
+def create_mechanics_functions(functionSpace, mode2D, materialModel, 
+                               pressureProjectionDegree=None,
+                               dt=0.0):
```

Comment:
Late comment. Can I ask, why did `dt` get moved into the factory function?

Reply:
If I remember correctly, things weren't properly connected so dt wasn't making its way to the material models. Maybe I misunderstood the interface as it was though?